File: GsonfromJson/app/src/main/java/com/example/gsonfromjson/MainActivity.java
package com.example.gsonfromjson;

import androidx.appcompat.app.AppCompatActivity;

import android.gesture.GestureOverlayView;
import android.os.Bundle;

import com.google.gson.Gson;
import com.google.gson.stream.JsonReader;

import java.text.AttributedString;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Gson gson = new Gson();

        String json = "{\"name\": \"Arman\", \"lName\" : \"Ziaei\" , \"age\" : 17 , \"job\" : \"Student\" }";
        Users users = gson.fromJson(json , Users.class);
    }
}

File: sample2/app/src/main/java/com/example/sample/MainActivity.kt
package com.example.sample

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.example.sample.automobile.Car
import com.example.sample.di.CarComponent
import javax.inject.Inject

class MainActivity : AppCompatActivity() {

    lateinit var component : CarComponent

    @Inject
    lateinit var car : Car

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        component = Dagger
    }
}